app/workers/fast_track/hypertension_observation_data.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module FastTrack
+  class HypertensionObservationData
+    attr_accessor :response
+
+    def initialize(response)
+      @response = response
+    end
+
+    def transform
+      entries = response.body['entry']
+      entries.map { |entry| transform_entry(entry) }
+    end
+
+    private
+
+    def pick(keys, hash)
+      hash.select { |k, _| keys.include? k }.with_indifferent_access
+    end
+
+    def transform_entry(raw_entry)
+      entry = pick(%w[issued component performer], raw_entry['resource'])
+      result = { issued: entry['issued'] }
+      practitioner_hash = get_display_hash_from_performer('Practitioner', entry)
+      organization_hash = get_display_hash_from_performer('Organization', entry)
+      bp_hash = get_bp_readings_from_entry(entry)
+      result.merge(practitioner_hash, organization_hash, bp_hash)
+    end
+
+    def get_display_hash_from_performer(term, entry)
+      result = {}
+      if entry['performer'].present?
+        performer_with_term = entry['performer'].select { |item| item['reference'].include? term }
```

Comment:
You can use `detect` here instead of selecting all of them and just asking for first one on the next line.